src/_pytest/config/findpaths.py

```diff
@@ -119,36 +173,23 @@ def determine_setup(
     rootdir_cmd_arg: Optional[str] = None,
     config: Optional["Config"] = None,
 ) -> Tuple[py.path.local, Optional[str], Any]:
```

Comment:
To top it off:

```suggestion
) -> Tuple[py.path.local, Optional[str], Dict[str, Union[str, List[str]]]]:
```

Reply:
Done (I had to change `Config._getini` slightly to make mypy happy)